### PR TITLE
feat(cli/repl): automatically indent multi-line inputs

### PR DESCRIPTION
**Pull request by caspervonb**
This adds auto-indentation based on brace balance level in the pending code to multi-line inputs of the repl.

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Can we get a test?

**Comment by caspervonb** [comment]
> Can we get a test?

Bit tricky since this is readline input but I think we could assert that the history file ends with expected input.

**Comment by ry** [comment]
Thanks for the patch Casper. This is a nice usability feature - I tried it manually - works well. But it's also extra complexity. I'm not sure if its worth the trade off without a test.

File: cli/repl.rs [hunk 1]
```diff
@@ -57,6 +57,24 @@ impl Repl {
       .map_err(AnyError::from)
 
     // Forward error to TS side for processing
+    // forward error to ts side for processing
```
Comment (caspervonb) [inline cli/repl.rs:60]: Oops, rogue `gu` movement.

```suggestion
    // Forward error to ts side for processing
```

File: cli/repl.rs [hunk 2]
```diff
@@ -57,6 +57,24 @@ impl Repl {
       .map_err(AnyError::from)
 
     // Forward error to TS side for processing
+    // Forward error to ts side for processing
```
Comment (caspervonb) [inline cli/repl.rs:60]:
```suggestion
```

**Comment by caspervonb** [comment]
> Thanks for the patch Casper. This is a nice usability feature - I tried it manually - works well. But it's also extra complexity. I'm not sure if its worth the trade off without a test.

Low priority, this is a nice to have feature but not required so I'll get to it once I've sorted out some of the higher priority items in my queue 👍

**Review by crypt096 — APPROVED** [review]

**Comment by caspervonb** [comment]
Superseded by #7763

**Pull request closed without merging**